kyaml/fn/runtime/runtimeutil/functiontypes.go

```diff
@@ -84,10 +84,18 @@ type StorageMount struct {
 
 	// The path where the file or directory is mounted in the container.
 	DstPath string `json:"dst,omitempty" yaml:"dst,omitempty"`
+
+	// Mount in ReadWrite mode if it's explicitly configured
+	// See https://docs.docker.com/storage/bind-mounts/#use-a-read-only-bind-mount
+	ReadWriteMode bool `json:"rw,omitempty" yaml:"rw,omitempty"`
```

Comment:
I checked if these tags are used:
Right now this structure is un-marshaled only when we read function annotation. And in that place this data is completely disregarded/ignored [1]. The only way to put the data in this structure that is used is via StringToStorageMount.
That means that those tags are not used anyway.
Do you think it's still potentially dangerous?

In general I think it won't hurt and I'm ok with removing that tags for the 'security'-risks reasons you mentioned even though right now there is no issues with that.
One more thought I had: it may be better to remove tags from StorageMounts and Network here [2], because we're ignoring that anyway in [1]. But it may break some tests. I'll play with it a bit more, but most likely you proposal is safer :)

[1]
https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/runtimeutil/functiontypes.go#L106

[2]
https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/runtimeutil/functiontypes.go#L38

Reply:
Sounds good.  If those tags aren't unmarshalled directly let's remove them to reduce confusion.  Can be done as a follow up.

Reply:
I've done the experiment:
There will be an issue if we remove that tags, yaml/json marshal/unmarshal functions still see the struct fields during unmarshalling from kpt-function annotation even without tags. E.g. [1] as you can see marshaling still works even without tags. Tags are needed just to have more granular control, like: rename fields or omit them if they're empty.
That means that if we want to guard, we need to do either:
1. create a parent struct for type FunctionSpec and move the fields we don't want to present in kpt-function annotation to it. and include there the current FunctionSpec that we'll need to unmarshal to read the data from annotations.
2.  just prune(clean data in) the not needed fields before marshaling and after unmarshaling.

In practice we use right now the 2nd approach:
we don't marshal FunctionSpec (actually, do it, but only in our tests)
we clean the data for unneeded fields after unmarshaling [2]

I suggest keeping the tags as-is for now, because it won't guard that fields anyway.
+ may be plan some refactoring as it's proposed in the option 1. But it will affect at least 2 other modules: RunFns and fnpluings in kustomize + many tests will be rewritten. 

Meanwhile this issue with mount doesn't allow to use kpt/kustomize in cases you need some input/output via fs.
e.g. I suspect that the RC of https://github.com/GoogleContainerTools/kpt/issues/587  is related to the current issue.


[1]
https://golang.org/pkg/encoding/json/#example_Marshal

[2]
https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/runtimeutil/functiontypes.go#L106

Reply:
>Leaving the tags sounds good for now.
resolving then

Reply:
I guess we do unmarshal function spec directly from the annotation in `getFunctionSpecFromAnnotation`, but clear the StorageMounts and network, which get set from the CLI flags.